File: Easy/009.palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if(x < 0):
            return False
        xStr = str(x)
        reverse_xStr = xStr[::-1]
        if(x == int(reverse_xStr)):
            return True
        else:
            return False

File: Easy/009.palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if( x < 0 || (x != 0 && x % 10 == 0)){
            return false;
        }
        
        int reverse = 0;
        while(x > reverse){
            reverse = x % 10 + reverse * 10;
            x /= 10;
        }
    
        return x == reverse || x == reverse/10;
        
    }
}